horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/UserServiceImpl.java

```diff
@@ -58,29 +55,41 @@ public class UserServiceImpl implements UserService {
         return backend.get().getTeams();
     }
 
-    @WithRoles(addUsername = true)
+    @Transactional
+    @WithRoles(extras = Roles.HORREUM_SYSTEM)
     @Override public String defaultTeam() {
         UserInfo userInfo = UserInfo.findById(identity.getPrincipal().getName());
         return userInfo != null ? userInfo.defaultTeam : null;
     }
 
     @Transactional
     @WithRoles(addUsername = true)
-    @Override public void setDefaultTeam(String team) {
+    @Override public void setDefaultTeam(String unsafeTeam) {
+        String team = unsafeTeam != null ? validateTeamName(unsafeTeam) : null;
```

Comment:
So this can set team to null, which will set userInfo.defaultTeam = null, is that intended?

Reply:
Would it not be better to do: String name = validateTeamName(unsafeTeam); ?

Reply:
both approaches are reasonable. it depends if we want to allow `defautTeam` to be unset or not.

Reply:
changed to not allow unsettling `defaultTeam`